### ISSUE TITLE
Cardinal VST3 appears now as instrument 

### ISSUE THREAD

**Issue by subderground** [DISTRHO/Cardinal#626]
Hi,
I allways install the "normal" Cardinal version (not Fx or synth). But I just noticed that the last version is labelled as VST3i. Before It was just VST3 ; but since it's a VST3i, the audio is routing thru event if there is no in/out connected inside Cardinal.
I have tried LV2 Cardinal version and this works normally (no cable connected -> no sound).   
Could you check that ?
Many thanks.
T

**Comment by rafaelferreiraql** [comment]
I just updated to 23.10 and am having this issue as well.

**Comment by rafaelferreiraql** [comment]
UPDATE: Ok I think I know what's going on now. From the 23.10 changelog,
> Set main variant as synth type
In Reaper, VST instruments mix with oncoming input by default (currently looking if theres any way around that). Personally I don't think that's too helpful, considering the "Main" version is the only one which allows using Audio 8, clearly made for utilizing incoming audio... Which one would usually not use as mixed with incoming input. Maybe that change makes more sense for other DAWs which tread fx and instruments more differently.

**Comment by subderground** [comment]
Yes, Cardinal Vst3 doesn't match with a real hardware at this point of view. Audio signal should be muted if not connected. 
My 2 cents.😉

**Comment by beguinne** [comment]
Audio still passes through the main variant VST3i as of 24.05. In REAPER you can get the behavior you want by adding extra channels for audio input into Cardinal and mapping them something like this in conjunction with a JS Channel Downmixer. But I agree it'd be simpler if sound only passed through if you patch it in the plugin.

![reaper_YQMolXIR4m](https://github.com/DISTRHO/Cardinal/assets/172810064/2eb5ae3e-8dab-48cc-a479-4361a36d87d6)

**Comment by rafaelferreiraql** [comment]
> Audio still passes through the main variant VST3i as of 24.05. In REAPER you can get the behavior you want by adding extra channels for audio input into Cardinal and connecting the pins something like this in conjunction with a JS Channel Mapper / Downmixer. But I agree it'd be simpler if sound only passed through once you patch it in the plugin.

Yea, the thing about the workaround is that it's already taking away two additional channels just for it. Given one of the main reasons someone would want to insist on using main Cardinal rather than Cardinal FX would be having access to extra channels, it just becomes more of a mess.

**Comment by beguinne** [comment]
Ah, so the audio channels of a track in REAPER are separate from the Cardinal plugin's input and output channels. You may add as many extra channels in REAPER as you like and map them to Cardinal's channels in such a way that your input audio dodges the problematic Cardinal channels. So you won't lose access to any of Cardinal's channels if you set it up something like this (there are a lot of ways to do it, but this is an example that shows both mono and stereo I/O): 

![cPG0ddRQN1](https://github.com/DISTRHO/Cardinal/assets/172810064/72c62bc7-4f88-442a-9f96-ead3eddfdedf)

![reaper_msUH3NMiTA](https://github.com/DISTRHO/Cardinal/assets/172810064/a5567aa3-5a5d-4113-969e-5037e5f0524d)

I've attached this example project here. But yeah I agree this is pretty messy just to avoid the issue of audio passing through.

[Cardinal input routing in REAPER.zip](https://github.com/user-attachments/files/15877771/Cardinal.input.routing.in.REAPER.zip)

**Comment by falkTX** [comment]
Note that this might be an issue with the VST3 version in particular, due to how buses work on VST3 being quite complex I might have the implementation wrong.

Try the CLAP and LV2 versions, they should have different behaviour. Particularly the LV2 one should work without issues I believe, because there is no fancy bus mechanisms there, all ports are exposed by default on LV2 side

**Comment by rafaelferreiraql** [comment]
I'm glad this is being looked into!
> Try the CLAP and LV2 versions, they should have different behaviour. Particularly the LV2 one should work without issues I believe, because there is no fancy bus mechanisms there, all ports are exposed by default on LV2 side

Well, a bit unrelated to this issue but I tried CLAP (extracted it on my personal vst folder as I would any other plugin) and got this unfortunate result.
![image](https://github.com/user-attachments/assets/4110614c-edaa-4ab4-89e9-c4dce8ced6bc)

(LV2 is recognized but straight up refuses to open, on my Reaper at least)
